.github/actions/go-test/action.yml

```
@@ -0,0 +1,78 @@
+# Copyright 2024 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'go-test'
+
+inputs:
+  go_version:
+    description: 'The version of Go to install and use.'
+    type: 'string'
+  go_version_file:
+    description: 'Path to the go.mod file to extract a version.'
+    type: 'string'
+    default: 'go.mod'
+  go_packages:
+    description: 'List of go packages to test.'
+    type: 'string'
+    default: './...'
+  go_proxy:
+    description: 'Proxy script to wrap go commands.'
+    type: 'string'
+    default: 'go'
+  go_test_timeout:
+    description: 'Timeout for Go tests.'
+    type: 'string'
+    default: '10m'
+  directory:
+    description: 'Directory to the root of the Go module.'
+    type: 'string'
+    default: '.'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Checkout'
+      uses: 'actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11' # ratchet:actions/checkout@v4
+
+    - name: 'Setup Go'
+      uses: 'actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491' # ratchet:actions/setup-go@v5
+      with:
+        go-version: '${{ inputs.go_version }}'
+        go-version-file: '${{ inputs.go_version_file }}'
+
+    - name: 'Test'
+      shell: 'bash'
+      working-directory: '${{ inputs.directory }}'
+      run: |-
+        ${{ inputs.go_proxy }} test \
```

This will handle any shell escaping for you (spaces, special characters, etc):

```suggestion
      working-directory: '${{ inputs.directory }}'
      env:
        GO_COMMAND: '${{ inputs.go_command }}'
      run: |-
        "${GO_COMMAND}" test \
```